### PR TITLE
CA-208537: vdi-copy between local SRs proposes unwanted ciphers

### PR DESCRIPTION
**Pull request by phusl**
Enable TLSv1.2 capability for sparse_dd.

Signed-off-by: Phus Lu <phus.lu@citrix.com>

### CODE REVIEW

**Comment by phusl** [comment]
DONT merge, just for review currently. /cc @thomassa

**Comment by phusl** [comment]
Basically verified, @thomassa , are you happy to start review it now :smiley:

**Comment by simonjbeaumont** [comment]
Review from @thomassa specifically requested. He's off sick today so assigning to him and adding the blocked label so he can pick them up on his return.

**Comment by simonjbeaumont** [comment]
The travis job is failing because the latest changes to the deps haven't been uploaded to EC2 yet... @jonludlam. The xen-git job is passing so I assert (from a CI point of view only) that this PR is OK.

**Comment by simonjbeaumont** [comment]
@thomassa has returned so I'll remove the blocked label.

File: ocaml/xapi/sparse_dd_wrapper.ml [hunk 1]
```diff
@@ -70,8 +70,11 @@ let dd_internal progress_cb base prezeroed infile outfile size =
 					"-machine";
 					"-src"; infile;
 					"-dest"; outfile;
-					"-size"; Int64.to_string size
-				] @ (if prezeroed then [ "-prezeroed" ] else []
+					"-size"; Int64.to_string size;
+					"-good-ciphersuites"; (match !Xapi_globs.ciphersuites_good_outbound with None -> "!EXPORT:RSA+AES128-SHA256" | Some s -> s) ;
```
Comment (thomassa) [inline ocaml/xapi/sparse_dd_wrapper.ml]: We must not have any hardcoded default values here.
This should be handled as in `ocaml/xapi/xapi_sync.ml`, with
```
(match !Xapi_globs.ciphersuites_good_outbound with
    | Some s -> s
    | None -> raise (Api_errors.Server_error (Api_errors.internal_error,["Vdi_copy found no good ciphersuites in Xapi_globs."]))
);
```

**Comment by thomassa** [comment]
The commit-message needs a brief description of the new behaviour.

**Comment by simonjbeaumont** [comment]
FYI, we can expect CI to pass on any future commits (not withstanding issues within the commits themselves) as the CI yum-repos have been brought up to date. In fact, I'll restart the Travis job which should now pass.

**Comment by phusl** [comment]
Thanks for @thomassa comments. I reworked the commit.

**Comment by thomassa** [comment]
Looks good now.
This can be merged when xapi-project/vhd-tool#26 goes in.

**Pull request merged**